src/sensai/vector_model.py

```diff
@@ -133,20 +127,113 @@ def withName(self, name: str):
         self.setName(name)
         return self
 
+    def getInputTransformer(self, cls: Type[DataFrameTransformer]):
+        for it in self._inputTransformerChain.dataFrameTransformers:
+            if isinstance(it, cls):
+                return it
+        return None
+
+    def setName(self, name):
+        self._name = name
+
+    def getName(self):
+        if self._name is None:
+            return "unnamed-%s-%x" % (self.__class__.__name__, id(self))
+        return self._name
+
+    def setFeatureGenerator(self, featureGenerator: Optional[FeatureGenerator]):
+        self.withFeatureGenerator(featureGenerator)
+
+    def getFeatureGenerator(self) -> Optional[FeatureGenerator]:
+        return self._featureGenerator
+
+
+class FittableModel(PredictorModel, ABC):
     @abstractmethod
-    def isRegressionModel(self) -> bool:
+    def fit(self, X: pd.DataFrame, Y: pd.DataFrame = None):
+        pass
+
+
+class RulesBasedModel(FittableModel, ABC):
+    """
+    Base class for models where the essential logic is rule based but the feature generators
```

Comment:
suggestion: ... the essential prediction logic is based on rules coded by humans and thus does not require fitting. However, the input generation process may use mechanisms such as feature generation or data transformation, which may require fitting.

Reply:
Done